### PR TITLE
Get the typescript build process to generate type declarations

### PR DESCRIPTION
**Pull request by tomclose**

### CODE REVIEW

File: packages/fmg-core/tsconfig.json [hunk 1]
```diff
@@ -2,12 +2,13 @@
   "compilerOptions": {
     /* Basic Options */
     "lib": ["es5", "es6"],
-    "allowJs": true,                        /* Allow javascript files to be compiled. */
```
Comment (andrewgordstewart) [inline packages/fmg-core/tsconfig.json]: There are still a bunch of `js` files that haven't been rewritten yet.

File: packages/fmg-core/tsconfig.json [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
+    "allowJs": false,                        /* Allow javascript files to be compiled. */
     "outDir": "built",                      /* Redirect output structure to the directory. */
     "module": "commonjs",                   /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "declaration": true,
```
Comment (andrewgordstewart) [inline packages/fmg-core/tsconfig.json]: 👍

File: packages/fmg-core/tsconfig.json (hunk 1 above)
Comment (tomclose) [reply packages/fmg-core/tsconfig.json]: I couldn't see any ... ?

Comment (andrewgordstewart) [reply packages/fmg-core/tsconfig.json]: ... you're right

**Comment by andrewgordstewart** [comment]
Somehow, though, this change results in no tests running.
```
~/Code/magmo/minimal_viable_force_move_games/packages/fmg-core on generate-types*
$ npm test

> fmg-core@0.0.4 test /Users/andrewstewart/Code/magmo/minimal_viable_force_move_games/packages/fmg-core
> npm run build && ./bin/test.sh


> fmg-core@0.0.4 build /Users/andrewstewart/Code/magmo/minimal_viable_force_move_games/packages/fmg-core
> ./bin/build.sh

Compiling ./contracts/ForceMoveGame.sol...
Compiling ./contracts/Migrations.sol...
Compiling ./contracts/Rules.sol...
Compiling ./contracts/State.sol...
Compiling ./test/test-game/contracts/CountingGame.sol...
Compiling ./test/test-game/contracts/CountingState.sol...


  0 passing (0ms)
```

**Pull request merged**